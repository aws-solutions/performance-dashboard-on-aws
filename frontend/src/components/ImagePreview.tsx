import React from "react";

type Props = {
  title: string;
  summary: string;
  file: File | undefined;
  summaryBelow: boolean;
  altText: string;
};

const ImagePreview = (props: Props) => {
  const { file, summaryBelow, summary, title, altText } = props;

  return (
    <div className="preview-container">
      <h2 className="margin-left-1 margin-top-1">{title}</h2>
      {!summaryBelow && (
        <p className="margin-left-1 margin-top-0 margin-bottom-3">{summary}</p>
      )}
      <div className="margin-left-1">
        <img src={file ? URL.createObjectURL(file) : ""} alt={altText}></img>
      </div>
      {summaryBelow && (
        <p className="margin-left-1 margin-top-3 margin-bottom-0">{summary}</p>
      )}
    </div>
  );
};

export default ImagePreview;
