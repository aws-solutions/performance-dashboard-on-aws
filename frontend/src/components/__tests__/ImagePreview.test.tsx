import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ImagePreview from "../ImagePreview";

const imageFile = {
  type: "image/png",
  name: "myphoto.png",
  size: 100,
} as File;

window.URL.createObjectURL = jest.fn();

test("renders the title and summary of the image preview component", async () => {
  const { getByText } = render(
    <ImagePreview
      title="test title"
      summary="test summary"
      file={imageFile}
      summaryBelow={false}
    />,
    { wrapper: MemoryRouter }
  );
  expect(getByText("test title")).toBeInTheDocument();
  expect(getByText("test summary")).toBeInTheDocument();
  expect(getByText("test summary").nextSibling).toContainHTML("img");
});

test("renders the image preview component with the summary below the chart", async () => {
  const { getByText } = render(
    <ImagePreview
      title="test title"
      summary="test summary"
      file={imageFile}
      summaryBelow={true}
    />,
    { wrapper: MemoryRouter }
  );
  expect(getByText("test title")).toBeInTheDocument();
  expect(getByText("test summary")).toBeInTheDocument();
  expect(getByText("test summary").previousSibling).toContainHTML("img");
});

test("image preview should match snapshot", async () => {
  const wrapper = render(
    <ImagePreview
      title="test title"
      summary="test summary"
      file={imageFile}
      summaryBelow={false}
    />,
    { wrapper: MemoryRouter }
  );
  expect(wrapper.container).toMatchSnapshot();
});
