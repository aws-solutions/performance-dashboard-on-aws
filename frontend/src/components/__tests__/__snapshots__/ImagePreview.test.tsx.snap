// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image preview should match snapshot 1`] = `
<div>
  <div
    class="preview-container"
  >
    <h2
      class="margin-left-1 margin-top-1"
    >
      test title
    </h2>
    <p
      class="margin-left-1 margin-top-0 margin-bottom-3"
    >
      test summary
    </p>
    <div
      class="margin-left-1"
    >
      <img
        alt="alt text"
      />
    </div>
  </div>
</div>
`;
